src/api.js

```diff
@@ -17,207 +17,221 @@ import currency from './currency';
-    reqHeaders['X-Locale'] = locale;
-  }
-
-  if (currency) {
-    reqHeaders['X-Currency'] = currency;
-  }
+    if (session) {
+      reqHeaders['X-Session'] = session;
+    }
 
-  const response = await fetch(reqUrl, {
-    method: reqMethod,
-    headers: reqHeaders,
-    body: reqBody,
-    // Credentials and mode are only available in the browser
-    ...(!utils.isServer()
-      ? {
-          credentials: 'include',
-          mode: 'cors',
-        }
-      : undefined),
-  });
+    if (locale) {
+      reqHeaders['X-Locale'] = locale;
+    }
 
-  const responseSession = response.headers.get('X-Session');
+    if (currency) {
+      reqHeaders['X-Currency'] = currency;
+    }
 
-  if (typeof responseSession === 'string' && session !== responseSession) {
-    allOptions.setCookie('swell-session', responseSession);
-  }
+    const response = await fetch(reqUrl, {
+      method: reqMethod,
+      headers: reqHeaders,
+      body: reqBody,
+      // Credentials and mode are only available in the browser
+      ...(!utils.isServer()
+        ? {
+            credentials: 'include',
+            mode: 'cors',
+          }
+        : undefined),
+    });
+
+    const responseSession = response.headers.get('X-Session');
+
+    if (typeof responseSession === 'string' && session !== responseSession) {
+      allOptions.setCookie('swell-session', responseSession);
+    }
 
-  // Response could be text, json, or empty
-  let result = null;
-  try {
-    result = await response.text();
+    // Response could be text, json, or empty
+    let result = null;
     try {
-      result = JSON.parse(result);
+      result = await response.text();
+      try {
+        result = JSON.parse(result);
+      } catch (err) {
+        // noop
+      }
     } catch (err) {
       // noop
     }
-  } catch (err) {
-    // noop
+
+    if (result && result.error) {
+      const err = new Error(result.error.message || result.error);
+      err.status = response.status;
+      err.code = result.error.code;
+      err.param = result.error.param;
+      throw err;
+    } else if (!response.ok) {
+      const err = new Error(
+        'A connection error occurred while making the request',
+      );
+      err.code = 'connection_error';
+      throw err;
+    }
+
+    return allOptions.useCamelCase ? utils.toCamel(result) : result;
   }
 
-  if (result && result.error) {
-    const err = new Error(result.error.message || result.error);
-    err.status = response.status;
-    err.code = result.error.code;
-    err.param = result.error.param;
-    throw err;
-  } else if (!response.ok) {
-    const err = new Error(
-      'A connection error occurred while making the request',
-    );
-    err.code = 'connection_error';
-    throw err;
+  if (initStore) {
+    api.init(initStore, initKey, initOpt);
   }
 
-  return allOptions.useCamelCase ? utils.toCamel(result) : result;
+  return api;
 }
 
-export default api;
+const instance = swell();
+
+instance.create = swell;
```

Comment:
`store` should be required for `create`, right?

Reply:
It's only required if `initStore` is provided, otherwise the user can call `init()` themselves, hence this:

```
if (initStore) {
  api.init(initStore, initKey, initOptions);
}
```